routers/private/hook_post_receive.go

```
@@ -124,6 +125,40 @@ func HookPostReceive(ctx *gitea_context.PrivateContext) {
 		refFullName := opts.RefFullNames[i]
 		newCommitID := opts.NewCommitIDs[i]
 
+		// post update for agit pull request
+		if git.SupportProcReceive && strings.HasPrefix(refFullName, git.PullPrefix) {
+			if repo == nil {
+				repo = loadRepository(ctx, ownerName, repoName)
+				if ctx.Written() {
+					return
+				}
+			}
+
+			pullIndexStr := strings.TrimPrefix(refFullName, git.PullPrefix)
+			pullIndexStr = strings.Split(pullIndexStr, "/")[0]
+			pullIndex, _ := strconv.ParseInt(pullIndexStr, 10, 64)
+			if pullIndex <= 0 {
+				continue
+			}
+
+			pr, err := models.GetPullRequestByIndex(repo.ID, pullIndex)
+			if err != nil {
```

If the PR doesn't exist, I think we could return an error response (maybe an error log is enough?), in case some PRs are really deleted in database.

And the below code also ignores the error if a PR doesn't exist.